### PR TITLE
#163518700  User can share an article on social media

### PR DESCRIPTION
**Pull request by niomwungeri-fabrice**
#### What does this PR do?
Share article on different social media including email
#### Description of Task to be completed?
Have this endpoints work
EndPoints:
* `GET /article/:articleId/share/twitter` for sharing on twitter.
* `GET /article/:articleId/share/fb` for sharing on facebook.
* `GET /article/:articleId/share/email` for sharing on email.
* `GET /article/:articleId/share/linkedIn` for sharing on linkedin.
#### How should this be manually tested?
* Fetch the branch `ft-user-can-share-163518700`
* Checkout the branch `ft-user-can-share-163518700`
* Setup your database. 
* Run tests `yarn run test`
* Start the server `yarn run server`
* Create a user `POST: /api/v1/users` with `email, password, and username`
* Login a user `POST: /api/v1/auth/login` with `email and password` you will get a token
* Pass token to header and try to access one of the above endpoint in PR description using `POSTMAN`
#### What are the relevant pivotal tracker stories?
163518700

### CODE REVIEW

File: controllers/articles.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    try {
+      const { tagsList = [], ...rest } = req.body;
+      const { id: userId } = req.user;
+      const refs = [];
+      const valid = await articleValidator(req.body);
+      const user = await User.findOne({ where: { id: userId } });
+      if (user && valid) {
+        const article = await Article.create({ ...rest, userId });
+        const { id: articleId } = article.dataValues;
+        // eslint-disable-next-line no-restricted-syntax
```
Comment (abayo-luc) [inline controllers/articles.js]: remove these comments by eslint.

**Review by abayo-luc — CHANGES_REQUESTED** [review]
we have two controllers doing exactly the same thing, ```./controller/articles.js``` and ```./controller/articlesController.js```

File: controllers/articles.js (hunk 1 above)
Comment (niomwungeri-fabrice) [reply controllers/articles.js]: @abayo-luc Done!

**Review by Musigwa-zz — CHANGES_REQUESTED** [review]
The whole work looks great however, I can't see the swagger documentation and the jsDoc comments.
Please consider documenting your API.

**Review by abayo-luc — APPROVED** [review]

File: controllers/articlesController.js [hunk 2]
```diff
@@ -49,4 +50,113 @@ export default class PostController {
       return res.status(500).send({ message: error, status: 500 });
     }
   }
+
```
Comment (puritybirir) [inline controllers/articlesController.js:53]: jsdoc comments

**Pull request closed without merging**